src/main/java/net/consensys/orion/http/handler/receive/ReceiveHandler.java

```diff
@@ -104,4 +99,27 @@ public void handle(final RoutingContext routingContext) {
       routingContext.response().end(toReturn);
     });
   }
+
+  private byte[] decryptPayload(
+      final RoutingContext routingContext,
+      final List<Box.PublicKey> recipients,
+      final EncryptedPayload encryptedPayload) {
+    byte[] decryptedPayload = null;
+    EnclaveException exception = null;
+    for (final Box.PublicKey recipient : recipients) {
```

Comment:
This would be simpler using a stream with map and findFirst

Reply:
That does not lead to nice code, as the decrypt throws exceptions ...